misc/openapi-ballerina/modules/openapi-validator/src/main/java/org/ballerinalang/openapi/validator/MatchResourcewithOperationId.java

```diff
@@ -0,0 +1,534 @@
+                    while (operationIter.hasNext()) {
+                        Map.Entry<String, Operation> operationMap = operationIter.next();
+                        if (!filters.getOperation().contains(operationMap.getValue().getOperationId())) {
+                            operationIter.remove();
+                        }
+                    }
+                }
+            } else if (excludeOperationFilteringEnable) {
+                // If exclude tags filtering available validate only the filtered exclude operations grouped by
+                // given exclude tags.
+                // Else If tags filtering available validate only the operations that filtered by exclude
+                // operations.
+                // Else proceed only to validate filtered exclude operations.
+                if (excludeTagsFilteringEnabled) {
+                    Iterator<Map.Entry<String, Operation>> operationIter =
+                            openAPIPathSummary.getOperations().entrySet().iterator();
+                    while (operationIter.hasNext()) {
+                        Map.Entry<String, Operation> operationMap = operationIter.next();
+                        if (!filters.getExcludeOperation().contains(operationMap.getValue().getOperationId())) {
+                            if (operationMap.getValue().getTags() != null) {
+                                if (!Collections.disjoint(filters.getExcludeTag(), operationMap.getValue().getTags())) {
+                                    operationIter.remove();
+                                }
+                            } else {
+                                operationIter.remove();
+                            }
+                        } else {
+                            operationIter.remove();
+                        }
+                    }
+                } else if (tagFilteringEnabled) {
+                    Iterator<Map.Entry<String, Operation>> operations =
+                            openAPIPathSummary.getOperations().entrySet().iterator();
+                    while (operations.hasNext()) {
+                        Map.Entry<String, Operation> operationMap = operations.next();
+                        if (!filters.getExcludeOperation().contains(operationMap.getValue().getOperationId())) {
+//                                    Check tag is available if it is null and not included in list
+//                                    then remove operations.
+                            if (operationMap.getValue().getTags() == null) {
+                                operations.remove();
+                            } else if (Collections.disjoint(filters.getTag(), operationMap.getValue().getTags())) {
+                                operations.remove();
+                            }
+                        } else {
+                            operations.remove();
+                        }
+                    }
+                } else {
```

Comment:
If we are doing the same operations for all places, Why we are having these checks separately? Please correct me, If I missed anything.